deploy/utils.go

```diff
@@ -202,12 +203,15 @@ func NewResourcesFromFile(filepath, namespace string) ([]Resource, error) {
 		return nil, err
 	}
 
-	return NewResourcesFromBuffer(stream, namespace)
+	return NewResourcesFromBuffer(stream, namespace, filepath)
 }
 
 // NewResourcesFromBuffer creates new Resources from a byte stream
 // Supports multiple resources divided by `---`
-func NewResourcesFromBuffer(stream []byte, namespace string) ([]Resource, error) {
+func NewResourcesFromBuffer(stream []byte, namespace string, filepath string) ([]Resource, error) {
+	if filepath == "" {
+		filepath = "buffer"
+	}
```

Comment:
I'm not liking this part very much, the function signature with filepath even if you are passing a buffer directly is not very clear.

We can revise this making the returned Resource with Filepath = "buffer" and override the values inside the NewResourcesFromFile function

Reply:
I added an internal function `createResourcesFromBuffer`, and used `NewResourcesFromBuffer` as a wrapper to set the filepath to `buffer` when generating the array of resources from a byte stream

Reply:
Nice catch! I added an internal function createResourcesFromBuffer, and used NewResourcesFromBuffer as a wrapper to set the filepath to buffer when generating the array of resources from a byte stream

Reply:
Nice catch! I added an internal function `createResourcesFromBuffer`, and used `NewResourcesFromBuffer` as a wrapper that sets `filepath` to `buffer` when generating the array of resources from a byte stream